signer/tests/integration/transaction_coordinator.rs

```diff
@@ -0,0 +1,331 @@
+    let signing_threshold = 5;
+    let context_window = 10;
+
+    let network = network::in_memory::Network::new();
+    let signer_info = testing::wsts::generate_signer_info(&mut rng, num_signers);
+
+    let mut testing_signer_set =
+        testing::wsts::SignerSet::new(&signer_info, signing_threshold, || network.connect());
+
+    let (aggregate_key, bitcoin_chain_tip) =
+        run_dkg(&context, &mut rng, &mut testing_signer_set).await;
+
+    // Ensure we have a signers UTXO (as a donation, to not mess with the current
+    // temporary `get_swept_deposit_requests` implementation)
+    push_utxo_donation(&context, &aggregate_key, &setup.sweep_block_hash).await;
+
+    assert_eq!(
+        context
+            .get_storage()
+            .get_swept_deposit_requests(&bitcoin_chain_tip.block_hash, context_window)
+            .await
+            .expect("failed to get swept deposits")
+            .len(),
+        1
+    );
+
+    let (broadcasted_transaction_tx, _broadcasted_transaction_rxeiver) =
+        tokio::sync::broadcast::channel(1);
+
+    // This task logs all transactions broadcasted by the coordinator.
+    let mut wait_for_transaction_rx = broadcasted_transaction_tx.subscribe();
+    let wait_for_transaction_task =
+        tokio::spawn(async move { wait_for_transaction_rx.recv().await });
+
+    // Setup the stacks client mock to broadcast the transaction to our channel.
+    context
+        .with_stacks_client(|client| {
+            client.expect_submit_tx().times(1..).returning(move |tx| {
+                let tx = tx.clone();
+                let txid = tx.txid();
+                let broadcasted_transaction_tx = broadcasted_transaction_tx.clone();
+                Box::pin(async move {
+                    broadcasted_transaction_tx
+                        .send(tx)
+                        .expect("Failed to send result");
+                    Ok(SubmitTxResponse::Acceptance(txid))
+                })
+            });
+        })
+        .await;
+
+    // Get the private key of the coordinator of the signer set.
+    let private_key = select_coordinator(&setup.sweep_block_hash.into(), &signer_info);
+
+    // TODO: should the tx coordinator evloop use its private key instead of `config().signer.private_key`?
+    let mut coordinator_ctx = context.clone();
+    coordinator_ctx.inner.config_mut().signer.private_key = private_key;
+
+    // Bootstrap the tx coordinator event loop
+    let tx_coordinator = transaction_coordinator::TxCoordinatorEventLoop {
+        context: coordinator_ctx,
+        network: network.connect(),
+        private_key,
+        context_window,
+        threshold: signing_threshold as u16,
+        signing_round_max_duration: Duration::from_secs(10),
+    };
+    let tx_coordinator_handle = tokio::spawn(async move { tx_coordinator.run().await });
+
+    // TODO: here signers use all the same storage, should we use separate ones?
+    let event_loop_handles: Vec<_> = signer_info
```

Comment:
Kinda. Ideally, they should all use separate `PgStore` objects pointing to separate postgres databases because it makes for a more convincing test. But this is pretty good as is, I'm convinced that the code that is being tested "works".

Reply:
Yeah makes sense, once all the pieces land on main we can add a "complete" integration test running all the signers separately.